src/components/CreateVirtualFacilityModal.vue

```diff
@@ -170,6 +176,29 @@ export default defineComponent({
     markFacilityIdTouched() {
       (this as any).$refs.facilityId.$el.classList.add('ion-touched');
     },
+    async isFacilityAlreadyExists() {
+      let facilitiesCount = 0;
+
+      try {
+        const resp = await FacilityService.fetchFacilities({
+          inputFields: {
+            facilityId: this.formData.facilityId
+          },
+          entityName: "FacilityAndProductStore",
+          viewSize: 1
+        })
+
+        if (!hasError(resp)) {
+          facilitiesCount = resp.data.docs.length
+        } else {
+          throw resp.data
+        }
+      } catch(err) {
+        logger.error(err)
+      }
+
+      return facilitiesCount > 0
```

Comment:
Instead we should define a variable `isFacilityExist` and toggle its value based on the api response.